### PR TITLE
Generalizes one tile smoothing from exoplanets to all noise maps

### PR DESCRIPTION
**Pull request by comma**
If parameter is set, single tiles without buddies in orthogonal directions are going to get squashed.
I went for orthogonal because allowing diagonals leaves odd configurations that don't look nice.

### CODE REVIEW

File: code/modules/random_map/noise/noise.dm [hunk 1]
```diff
@@ -167,4 +168,33 @@
...
+				for(var/cell in neighbors)
+					if(noise2value(map[cell]) == noise2value(map[mapcell]))
+						buddies |= cell
+				if(!length(buddies))
+					map[mapcell] = map[pick(neighbors)]
+
+/datum/random_map/noise/proc/get_neighbors(x, y, include_diagonals)
+	. = list()
+	if(!include_diagonals)
+		for(var/list/offset in list(list(-1, 0), list(1, 0), list(0, 1), list(0,-1)))
```
Comment (Lohikar) [inline code/modules/random_map/noise/noise.dm]: How hot is this proc? Might be good to break this into a static local var just so you don't need to init 5 lists every time.

File: code/modules/random_map/noise/noise.dm [hunk 2]
```diff
@@ -167,4 +168,33 @@
...
+			var/tmp_cell = get_map_cell(x+offset[1],y+offset[2])
+			if(tmp_cell)
+				. += tmp_cell
+	else
+		var/list/offsets = list(-1,0,1)
```
Comment (Lohikar) [inline code/modules/random_map/noise/noise.dm]: This should be static (or even just a for-in loop), I don't see it being modified anywhere.

Comment (comma) [reply code/modules/random_map/noise/noise.dm]: Pushed

File: code/modules/random_map/noise/noise.dm (hunk 1 above)
Comment (comma) [reply code/modules/random_map/noise/noise.dm]: Done

**Review by MistakeNot4892 — DISMISSED** [review]

**Comment by MistakeNot4892** [comment]
Conflicts.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**